### ISSUE TITLE
how to debug a failed start?

### ISSUE THREAD

**Issue by fommil** [Tapad/sbt-docker-compose#70]
I'm afraid I don't know how to debug this failure to run `dockerComposeUp`, what can I do to try and find the problem?

I guess this is being called? https://github.com/Tapad/sbt-docker-compose/blob/master/src/main/scala/com/tapad/docker/DockerComposePlugin.scala#L269

I have `dockerImageCreationTask := (publishLocal in Docker).value` and this is published.

The `.yml` file is deleted before I can look at it.

It's kind of weird that it says it's skipping `ruleengine` (my project) and then it tries to pull it.

```
Reading Compose File: docker/ruleengine.yml
Created Compose File with Processed Custom Tags: /tmp/compose-updated7013807111702461847.yml
Pulling Docker images except for locally built images and images defined as <skipPull> or <localBuild>.
Skipping Pull of image: ruleengine:latest
Creating network "935436_default" with the default driver
Pulling ruleengine (ruleengine:latest)...
repository ruleengine not found: does not exist or no pull access
No stopped containers
935436_default
Error starting Docker Compose instance. Shutting down containers...
```

**Comment by kurtkopchik** [comment]
Hi @fommil - It would be helpful to see the contents of the `ruleengine.yml` file as well as the full output from `dockerComposeUp`.

By default the plugin uses the name of the sbt `project` to determine which `service` in the yml file is the one being built by the current project. If they differ you need to explicitly set the `composeServiceName` to be the name of `yml` service that contains the image being built.

` composeServiceName := // Specify the name of the service in the Docker Compose file being tested. This setting prevents the service image from being pull down from the Docker Registry. It defaults to the sbt Project name.`

Lastly you should make sure that `ruleengine:latest` actually exists in the available local docker images. If not, we would need to investigate if `(publishLocal in Docker).value` is creating the expected `ruleengine:latest` image.

**Comment by fommil** [comment]
it's ok, I got to the bottom of it out of random trial and error. We had `dockerRepository` in `sbt-native-packager` and this wasn't picked up by `sbt-docker-compose`.

I've concluded that we need only a very small subset of the features of this plugin and we need them as `Task`s so I'm going to just write something small and use that instead. I can ping you if we make it libre.

Thanks for the inspiration!
